docs/specs.md

```diff
@@ -87,181 +113,323 @@ can be considered as _a collection of text chunks_. How those chunks are
+with self-contained data; the exact shape of a chunk depends on the document
+model type.
+
+### Document header
+
+The document header contains metadata applicable to the full document. It is
+divided into sections, each one containing dictionary-like metadata.
+
+The following sections are defined:
+ * `document`: contains relevant information (metadata) describing this
+   particular document; a non-exahustive list of possible fields (all of them
+   optional except `id`) is:
+    - `id`: an arbitrary string acting as an identifier for the document. If
+	  the document belongs to a dataset, it should be unique across all
+	  documents in the dataset.
+    - `type`: the model type for the document (_sequence_, _tree_, _table_)
+      The default value, when the field is not present, is _sequence_
+    - `date`: an ISO 8601 date defining the creation date of the document
+    - `main_lang`: an ISO 639-1 code describing the language the document is
+	  in.
+      For documents containing more than one language, it should express the
+      most frequent language in the document.
+    - `other_lang`: a list of ISO 639-1 codes indicating other languages
+	  present in the document
+    - `country`: a list of one or more ISO 3166-1 country codes for which the
+      document may be applicable (this might be used to further define the
+      applicability of country-dependent PII elements)
+    - `title`: a title for the document
+ * `dataset`: metadata describing the whole dataset this document belongs to
+   (so their fields are applicable to _all_ documents in the dataset).
+
+Additional sections may be defined for specific document types.
+
+
+## Iteration interfaces for source documents
+
+Once the Source Document has been specified, we can define program interfaces
+to _iterate_ over a document, producing document chunks. Two different iteration
+interfaces have been defined for this data structure:
+ * **structural iteration**: this is the native iteration for the document,
+   which produces elements revealing its intrinsic data structure (therefore the
+   objects delivered in the iteration may differ across document types,
+   _sequence_, _table_, _tree_)
+ * **full iteration** (or flat iteration): this iteration interface flattens the
+   document structure and always delivers a linear sequence of chunks,
+   regardless of the document type.
+
+An additional difference between the two iteration modes appears in the 
+content of the contextual information delivered within each chunk:
+ * In structural iteration, the context fields for a chunk will not contain any
+   information regarding the rest of the document (since the structure is
+   already given by the iteration), only semantic information inherent to the
+   chunk
+ * A chunk context in full iteration, however, may contain context fields that
+   refer to other chunks or to the full document (depending on its structure).
+
```

Comment:
Theoretically yes. In practice I set the interfaces as they came out from the code :-). 

I'm not saying is not possible to combine, but in the structural iteration the definition of what the context contains is trickier, because chunks have more complicated relationships among them.

Reply:
Thank you